tests/h/traversal/contexts_test.py

```diff
@@ -162,6 +163,52 @@ def test_acl_flag_shared(self,
         else:
             assert not pyramid_request.has_permission('flag', res)
 
+    @pytest.mark.parametrize('groupid,userid,permitted', [
+        ('freeforall', 'jim', True),
+        ('freeforall', 'saoirse', True),
+        ('freeforall', None, False),
+        ('only-saoirse', 'jim', False),
+        ('only-saoirse', 'saoirse', True),
+        ('only-saoirse', None, False),
+        ('pals', 'jim', True),
+        ('pals', 'saoirse', True),
+        ('pals', 'francis', False),
+        ('pals', None, False),
+        ('unknown-group', 'jim', False),
+        ('unknown-group', 'saoirse', False),
+        ('unknown-group', 'francis', False),
+        ('unknown-group', None, False),
+    ])
+    def test_acl_moderate_shared(self,
+                                 factories,
+                                 pyramid_config,
+                                 pyramid_request,
+                                 groupid,
+                                 userid,
+                                 permitted,
+                                 group_service,
+                                 links_service):
+        """
+        Moderate permissions should only be applied when an annotation
+        is shared—as the annotation here is shared, anyone set as a principal
+        for the given ``FakeGroup`` will receive the ``moderate`` permission.
+        """
+        # Set up the test with a dummy authn policy and a real ACL authz
+        # policy:
+        policy = ACLAuthorizationPolicy()
+        pyramid_config.testing_securitypolicy(userid)
+        pyramid_config.set_authorization_policy(policy)
+
+        ann = factories.Annotation(shared=True,
+                                   userid='mioara',
+                                   groupid=groupid)
+        res = AnnotationContext(ann, group_service, links_service)
+
+        if permitted:
+            assert pyramid_request.has_permission('flag', res)
+        else:
+            assert not pyramid_request.has_permission('flag', res)
```

Comment:
Typo I presume. This should be `moderate` right? And for the line above.

Reply:
Willfix